ipm_cmd_vel_node:
  ros__parameters:

    enable_vis: true
    
    # ----------------------------------------------------
    # 1. 话题设置 (Topics)
    # ----------------------------------------------------
    # 订阅的原始/畸变校正后的图像话题
    image_topic: "/camera/color/image_raw" 
    
    # 发布的鸟瞰图障碍物点云话题 (给 Costmap)
    pointcloud_topic: "/bev/obstacles"
    
    # (可选) 发布的调试用鸟瞰图图像话题
    bev_debug_image_topic: "/bev/debug_image"

    # === 相机标定 ===
    camera_matrix: [560.08, 0.0, 313.60, 
                    0.0, 560.16, 226.29, 
                    0.0, 0.0, 1.0]
    distortion_coeffs: [0.1248, -0.1525, -0.0104, -0.0031, 0.0]


    # ----------------------------------------------------
    # 2. 鸟瞰图 (BEV) 尺寸
    # ----------------------------------------------------
    # 输出的鸟瞰图的像素尺寸 (高和宽)
    bev_width: 640
    bev_height: 480

    # ######最初小矩形
    # src_points: [201.0, 357.0, 442.0, 356.0, 558.0, 447.0, 89.0, 452.0]
    # dst_points: [0.0, 0.0, 640.0, 0.0, 640.0, 480.0, 0.0, 480.0]
    # world_width_m: 0.244
    # world_height_m: 0.273
    # origin_offset_x_m: 0.325  # BEV "最近边" 在车前多远
    # origin_offset_y_m: 0.122  # BEV "左侧边" 在车左多远


    # # --- 配置 1 (例如：近场) ---
    # bev_width_1: 640
    # bev_height_1: 480
    # src_points_1: [201.0, 357.0, 442.0, 356.0, 558.0, 447.0, 89.0, 452.0]
    # dst_points_1: [0.0, 0.0, 640.0, 0.0, 640.0, 480.0, 0.0, 480.0]
    # world_width_m_1: 0.244
    # world_height_m_1: 0.273
    # origin_offset_x_m_1: 0.325
    # origin_offset_y_m_1: 0.122


    # # --- 配置 4 (例如：cmd_vel场) ---
    # bev_width_4: 640
    # bev_height_4: 480
    src_points_1: [210.0, 340.0, 429.0, 341.0, 611.0, 467.0, 49.0, 469.0]
    dst_points_1: [0.0, 0.0, 640.0, 0.0, 640.0, 480.0, 0.0, 480.0]
    world_width_m_1: 0.265
    world_height_m_1: 0.5
    origin_offset_x_m_1: 0.359
    origin_offset_y_m_1: 0.1325

    
    hsv_lower_red1_1: [0, 90, 50]
    hsv_upper_red1_1: [10, 255, 255]
    hsv_lower_red2_1: [170, 80, 50]
    hsv_upper_red2_1: [180, 255, 255]
    
    # --- 红色检测阈值 2 (用于 BEV 2) ---
    hsv_lower_red1_2: [0, 90, 50]
    hsv_upper_red1_2: [10, 255, 255]
    hsv_lower_red2_2: [170, 80, 50]
    hsv_upper_red2_2: [180, 255, 255]

    
    # ----------------------------------------------------
    # 7. 图像处理 (可选调优)
    # ----------------------------------------------------
    # 形态学操作核大小 (用于去噪点和填补孔洞)
    morph_kernel_size: 5